### ISSUE TITLE
Keyboard doesn't clear out last character from KeyboardInputField

### ISSUE THREAD

**Issue by jburfield92** [Microsoft/MixedRealityToolkit-Unity#917]
I have a textbox with the KeyboardInputField script on it with the Keyboard prefab in the scene.

When I tap in the textbox and start inputting characters and then decide to backspace until the textbox is cleared out, the last character (which is the first character inputted) is not cleared from the textbox. Although, The textbox in the Keyboard prefab is cleared entirely. When I add a new character, it replaces the uncleared character from the textbox.

I was able to resolve the problem by removing the if statement from the following code in KeyboardInputField:

```
private void Keyboard_onTextUpdated(string newText)
{
    if (!string.IsNullOrEmpty(newText))
    {
        this.text = newText;
    }
}  
```

**Comment by jasonhbartlett** [comment]
I don't know if this is the fix but I was able to deal with it by replacing that with:
`        private void Keyboard_OnTextUpdated(string newText)
        {
            text = Keyboard.Instance.InputField.text;
            //if (!string.IsNullOrEmpty(newText))
            //{
            //    text = newText;
            //}
        }`
Of course that makes the parameter newText obsolete, but...

**Comment by davidkline-ms** [comment]
This was released in the 2017.4.0.0 release candidate
